src/plugins/usage_collection/public/components/track_application_view/track_application_view_component.tsx

```diff
@@ -15,26 +15,34 @@ interface Props extends TrackApplicationViewProps {
   applicationUsageTracker?: IApplicationUsageTracker;
 }
 
-export class TrackApplicationViewComponent extends Component<Props> {
-  onClick = () => {
+export class TrackApplicationViewComponent extends React.Component<Props> {
+  private parentNode: (Node & ParentNode) | null | undefined;
+
+  onClick = (e: MouseEvent) => {
     const { applicationUsageTracker, viewId } = this.props;
-    applicationUsageTracker?.updateViewClickCounter(viewId);
+    this.parentNode = this.parentNode || ReactDOM.findDOMNode(this)?.parentNode;
+    if (this.parentNode === e.target || this.parentNode?.contains(e.target as Node | null)) {
```

Comment:
I'm not sure if the click count events increase more than they should. 
<details>
<summary>After a few clicks to create an ml job, we have:</summary>

```json
"ml": {
              "appId": "ml",
              "viewId": "main",
              "clicks_total": 2,
              "clicks_7_days": 2,
              "clicks_30_days": 2,
              "clicks_90_days": 2,
              "minutes_on_screen_total": 0.47473333333333334,
              "minutes_on_screen_7_days": 0.47473333333333334,
              "minutes_on_screen_30_days": 0.47473333333333334,
              "minutes_on_screen_90_days": 0.47473333333333334,
              "views": [
                {
                  "appId": "ml",
                  "viewId": "/",
                  "clicks_total": 1,
                  "clicks_7_days": 1,
                  "clicks_30_days": 1,
                  "clicks_90_days": 1,
                  "minutes_on_screen_total": 0.00023333333333333333,
                  "minutes_on_screen_7_days": 0.00023333333333333333,
                  "minutes_on_screen_30_days": 0.00023333333333333333,
                  "minutes_on_screen_90_days": 0.00023333333333333333
                },
                {
                  "appId": "ml",
                  "viewId": "/jobs/new_job/step/index_or_search",
                  "clicks_total": 1,
                  "clicks_7_days": 1,
                  "clicks_30_days": 1,
                  "clicks_90_days": 1,
                  "minutes_on_screen_total": 0.038016666666666664,
                  "minutes_on_screen_7_days": 0.038016666666666664,
                  "minutes_on_screen_30_days": 0.038016666666666664,
                  "minutes_on_screen_90_days": 0.038016666666666664
                },
                {
                  "appId": "ml",
                  "viewId": "/jobs/new_job/step/job_type",
                  "clicks_total": 1,
                  "clicks_7_days": 1,
                  "clicks_30_days": 1,
                  "clicks_90_days": 1,
                  "minutes_on_screen_total": 0.0521,
                  "minutes_on_screen_7_days": 0.0521,
                  "minutes_on_screen_30_days": 0.0521,
                  "minutes_on_screen_90_days": 0.0521
                },
                {
                  "appId": "ml",
                  "viewId": "/overview",
                  "clicks_total": 3,
                  "clicks_7_days": 3,
                  "clicks_30_days": 3,
                  "clicks_90_days": 3,
                  "minutes_on_screen_total": 0.7559666666666667,
                  "minutes_on_screen_7_days": 0.7559666666666667,
                  "minutes_on_screen_30_days": 0.7559666666666667,
                  "minutes_on_screen_90_days": 0.7559666666666667
                }
              ]
            }
          },
```
</details>

When navigating back to ml from advanced settings, just landing on the ml overview page now registers application_usage metrics as:
<details>
<summary>After navigating to ml and then back to advanced Settings:</summary>

```json
"ml": {
              "appId": "ml",
              "viewId": "main",
              "clicks_total": 18,
              "clicks_7_days": 18,
              "clicks_30_days": 18,
              "clicks_90_days": 18,
              "minutes_on_screen_total": 2.22435,
              "minutes_on_screen_7_days": 2.22435,
              "minutes_on_screen_30_days": 2.22435,
              "minutes_on_screen_90_days": 2.22435,
              "views": [
                {
                  "appId": "ml",
                  "viewId": "/",
                  "clicks_total": 1,
                  "clicks_7_days": 1,
                  "clicks_30_days": 1,
                  "clicks_90_days": 1,
                  "minutes_on_screen_total": 0.00023333333333333333,
                  "minutes_on_screen_7_days": 0.00023333333333333333,
                  "minutes_on_screen_30_days": 0.00023333333333333333,
                  "minutes_on_screen_90_days": 0.00023333333333333333
                },
                {
                  "appId": "ml",
                  "viewId": "/jobs/new_job/step/index_or_search",
                  "clicks_total": 1,
                  "clicks_7_days": 1,
                  "clicks_30_days": 1,
                  "clicks_90_days": 1,
                  "minutes_on_screen_total": 0.038016666666666664,
                  "minutes_on_screen_7_days": 0.038016666666666664,
                  "minutes_on_screen_30_days": 0.038016666666666664,
                  "minutes_on_screen_90_days": 0.038016666666666664
                },
                {
                  "appId": "ml",
                  "viewId": "/jobs/new_job/step/job_type",
                  "clicks_total": 1,
                  "clicks_7_days": 1,
                  "clicks_30_days": 1,
                  "clicks_90_days": 1,
                  "minutes_on_screen_total": 0.0521,
                  "minutes_on_screen_7_days": 0.0521,
                  "minutes_on_screen_30_days": 0.0521,
                  "minutes_on_screen_90_days": 0.0521
                },
                {
                  "appId": "ml",
                  "viewId": "/overview",
                  "clicks_total": 3,
                  "clicks_7_days": 3,
                  "clicks_30_days": 3,
                  "clicks_90_days": 3,
                  "minutes_on_screen_total": 0.7559666666666667,
                  "minutes_on_screen_7_days": 0.7559666666666667,
                  "minutes_on_screen_30_days": 0.7559666666666667,
                  "minutes_on_screen_90_days": 0.7559666666666667
                },
                {
                  "appId": "ml",
                  "viewId": "/jobs/new_job/single_metric",
                  "clicks_total": 6,
                  "clicks_7_days": 6,
                  "clicks_30_days": 6,
                  "clicks_90_days": 6,
                  "minutes_on_screen_total": 1.3631333333333333,
                  "minutes_on_screen_7_days": 1.3631333333333333,
                  "minutes_on_screen_30_days": 1.3631333333333333,
                  "minutes_on_screen_90_days": 1.3631333333333333
                }
              ]
            }
          },
```
</details>

Reply:
True. This is caused by another bug with `incrementBy` on the server side changing `0` to `1` by mistake. I've added a fix in this PR to this as well